### PR TITLE
chore(deps-dev): update build to node 14

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

Updates build to use node 14 + associated typings. 

Only required change was updating `mock-adb` to account for a behavior change where it's now an error to pass a number as content to `fs.writeFileSync` (it was doing this for a pidFile).

Note that this intentionally does not bump the engine requirements in package.json (that is, the project is cross-compatible with node 12 and 14). That's because service is currently blocked from updating to node 14, and I didn't want to introduce undue friction for working across the service and web repos. The package.json engines entries should be updated to enforce using 14 once service can update.

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by dbjorge** [comment]
Linux web e2es are blocked waiting for a node-14 based playwright-docker image, tracked by https://github.com/microsoft/playwright/pull/4262

(we could install our own node on top of the image, but since they're updating soon enough to LTS anyway, seems more apt to just wait for them)

**Comment by dbjorge** [comment]
Windows/mac web e2e failures seem to be known/existing variants

Haven't diagnosed unified e2e failures yet

File: docs/building-web.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ This document describes how to build and test Accessibility Insights for Web (th
 
 You will need the following tools installed:
 
--   [Node](https://nodejs.org) >= 12.16.3 (check by running `node --version`) - This is the version being enforced on our builds
--   [Yarn](https://yarnpkg.com/getting-started/install) >= 1.22.4 (check by running `yarn --version`)
+-   [Node](https://nodejs.org) >= 14.15.0 (check by running `node --version`) - This is the version being enforced on our builds
```
Comment (peterdur) [inline docs/building-web.md:16]: Should we defer/adjust this change to the instructions until we update `package.json`? We can still build with 12.16.3.

Comment (dbjorge) [reply docs/building-web.md:16]: I wanted to encourage the use of the same version CI is building against, less room for error that way

**Review by peterdur — APPROVED** [review]

**Pull request merged**